src/preset_time.jl

```diff
@@ -24,28 +24,18 @@ function PresetTimeCallback(tstops, user_affect!;
         initialize = SciMLBase.INITIALIZE_DEFAULT,
         filter_tstops = true,
         sort_inplace = false, kwargs...)
-
-    # define an intrinsic version of insorted
-    # extend a new method to avoid type instability of condition (anonymous function)
-    _insorted(x::Number, y::Number) = x == y
-    _insorted(x, v) = insorted(x, v)
-
-    if tstops isa AbstractVector
-        if sort_inplace
-            sort!(tstops)
-        else
-            tstops = sort(tstops)
-        end
-    elseif !(tstops isa Number)
+    if !(tstops isa AbstractVector) && !(tstops isa Number)
         throw(ArgumentError("tstops must either be a number or a Vector. Was $tstops"))
     end
 
+    tstops = tstops isa Number ? [tstops] : (sort_inplace ? sort!(tstops) : sort(tstops))
+
     condition = let
         function (u, t, integrator)
             if hasproperty(integrator, :dt)
-                _insorted(t, tstops) && (integrator.t - integrator.dt) != integrator.t
+                insorted(t, tstops) && !iszero(integrator.dt)
```

Comment:
that's not the same.

Reply:
Why not? You can remove integrator.t on both sides of the equation, right?

Reply:
No. `(integrator.t - integrator.dt) == integrator.t`does not imply that `iszero(integrator.dt)`

Reply:
The whole point of this is to capture tstops for which the difference is below machine epsilon, i.e. `eps(integrator.t)`, and collapsing those. If you're not taking into account floating point truncation error then it will lead to bugs in that case.